src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-list/user-list.component.ts

```diff
@@ -104,12 +112,17 @@ export class UserListComponent implements OnInit {
         cellTransformation: CellTemplate.checkIcon
       },
       {
-        name: $localize`Password expiration date`,
+        name: $localize`Password expires`,
         prop: 'pwdExpirationDate',
         flexGrow: 1,
-        pipe: this.cdDatePipe
+        cellTemplate: this.durationTpl
       }
     ];
+    const settings: string[] = ['USER_PWD_EXPIRATION_WARNING_1', 'USER_PWD_EXPIRATION_WARNING_2'];
+    this.settingsService.getValues(settings).subscribe((data) => {
+      this.expirationWarningAlert = data['USER_PWD_EXPIRATION_WARNING_1'];
+      this.expirationDangerAlert = data['USER_PWD_EXPIRATION_WARNING_2'];
+    });
```

Comment:
One nice way Angular encourage to do things is by dealing with subscriptions/asyncs directly in the HTML template via [Async Pipe (`async`)](https://angular.io/api/common/AsyncPipe):
```ts
this.settings$ = this.settingsService.getValues(settings);
this.now = Date.now();
```

```html
<div *ngIf="settings$ | async as settings">
	  <i [ngClass]="{
		  'warning-icon': (settings.USER_PWD_EXPIRATION_WARNING_1 - now) > warning_threshold,
	  	  'error-icon': (settings.USER_PWD_EXPIRATION_WARNING_2 - now) > error_threshold,
	    }">
	  </i>
    ...
</div>
``` 
On the other hand, that code starts to look like a good candidate for encapsulation... And there is where pipes come into play (we already use pipes for styling things: have a look at shared/pipes/health-color.pipe.ts).

Reply:
I've changed this in 28a88d0, what do you think ? I think it's more readable but I don't know that much about efficiency here so if that's no good then I'll try to do it your way.

Reply:
While Angular allows this, I'm not a big fan of the TS code returning CSS class names. CSS class names belong to the presentation domain, hence to the HTML template.

Let me put it this way: just by looking at the HTML template you should be able to completely understand what's the final look of the component (additionally, as a maintainer you'll often `grep` for CSS class names in HTML only files, not expecting that CSS class names come from TS code).

So, instead of this, what about something like:
```html
<i [ngClass]="{'warning': userPassToExpireWarning, 'error': userPassToExpireError}" >
```
or the equivalent:
```html
<i [class.warning]="userPassToExpireWarning" [class.error]="userPassToExpireError" >
```

Both properties `userPassToExpireWarning` and `userPassToExpireError` are just booleans, so the TS only has to deal with the decision of whether users' passwords are soon to expire, but not with the name of the class used for the expiration.

However, in the end it's still very close to having something like:
```html
<i 
[class.warning]="remainingDays <= this.expirationWarningAlert && remainingDays > this.expirationDangerAlert"
[class.error]="remainingDays <= this.expirationDangerAlert"
>
```

Reply:
Okay I understand now. Thanks for the lenghty explanation :)

Reply:
Happy to do so!